src/Faithlife.Tracing/ITrace.cs

```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace Faithlife.Tracing
+{
+	public interface ITrace : IDisposable
+	{
+		// TODO: recording an error?
+		// TODO: recording status information at end of request
+		void Annotate(string key, string value);
+		void Inject(Action<string, string> inject);
```

Comment:
`Save` ?

Should this be moved to `ITraceFactory.SaveTrace` ?

Reply:
Yes, it should move to `ITracer.InjectTrace` (or `InjectTraceSpan`? getting a little verbose). I think we can keep the "extract/inject" terminology used by the Open Tracing specification.

Reply:
I'm personally fine leaving off "span". I find "extract" and "inject" annoying because (for some reason) it isn't immediately obvious to me which direction it is going. (Am I injecting into the trace? or injecting into this thing?) But it's fine; just a bit of extra domain knowledge.